### ISSUE TITLE
What are the valid values of ldap_mech ?

### ISSUE THREAD

**Issue by williamdes** [cyrusimap/cyrus-sasl#631]
While creating https://github.com/tomav/docker-mailserver/pull/1708 I could not figure out what are the `ldap_mech` valid values.
Except some that can be found on the net:
- [DIGEST-MD5](https://docs.mongodb.com/manual/tutorial/configure-ldap-sasl-activedirectory/)
- [GSSAPI](https://pagure.io/SSSD/sssd/issue/863)
- [OTP](https://www.cyrusimap.org/sasl/sasl/faqs/openldap-sasl-gssapi.html)
- [EXTERNAL](https://serverfault.com/q/834523/336084)
- [CRAM-MD5](https://serverfault.com/q/834523/336084)
- [NTLM](https://serverfault.com/q/834523/336084)
- [LOGIN](https://serverfault.com/q/834523/336084)
- [PLAIN](https://serverfault.com/q/834523/336084)

https://github.com/cyrusimap/cyrus-sasl/blob/eeb935a9198172aede242c77b0e0dafe9312db10/lib/staticopen.h

Your documentation is quite empty for this option: https://github.com/cyrusimap/cyrus-sasl/blob/e41cfb986c1b1935770de554872247453fdbb079/docsrc/sasl/pwcheck.rst

And man pages are not better, I blame nobody but I want to have the valid values list and some details is possible.

Seems to have a list here: https://ldapwiki.com/wiki/SASL%20Mechanisms

Just found the command [here](https://unix.stackexchange.com/a/314856/155610) to get the server let us know it's values:
```
$ ldapsearch -ZZ -x -H ldap://ldap.example.com -b "" -LLL -s base supportedSASLMechanisms
dn:
supportedSASLMechanisms: SCRAM-SHA-1
supportedSASLMechanisms: SCRAM-SHA-256
supportedSASLMechanisms: GS2-IAKERB
supportedSASLMechanisms: GS2-KRB5
supportedSASLMechanisms: GSSAPI
supportedSASLMechanisms: GSS-SPNEGO
supportedSASLMechanisms: DIGEST-MD5
supportedSASLMechanisms: OTP
supportedSASLMechanisms: NTLM
supportedSASLMechanisms: CRAM-MD5
supportedSASLMechanisms: PLAIN
supportedSASLMechanisms: LOGIN

```

**Comment by williamdes** [comment]
SASL supported password schemes seem to be defined here:
https://github.com/cyrusimap/cyrus-sasl/blob/9b914e1d25d6414a2e4ef4c41a0a4ba27f20e79b/saslauthd/lak.c#L128-L136

This is in sync with my testing, any other value does not work

**Comment by quanah** [comment]
closed via #737
